### PR TITLE
DRUD-607: Passing vault token instead of path to file.

### PR DESCRIPTION
**Pull request by cyberswat**
## What this PR does / why we need it:
This PR removes the reliance on a .drud-sanctuary-token file.  Instead the token is managed via a yaml file (eg: ~/drud.yaml).  This helps centralize variable management.

## Link to JIRA issue this PR fixes:
https://newmediadenver.atlassian.net/browse/DRUD-607

## Notes and testing instructions for your reviewer:
From the perspective of this repository. the value of the vault token is already known.  This change is necessary in the bootstrap repository to support the same branch name.  I'll check the changes here into the vendor directory of that pull request to make testing easier for both repositories.  I was able to test these changes by walking through authenticating via `drud auth github`.  I used `drud secret list /` to validate authentication.

## Release/Deployment notes:
* Once https://github.com/drud/bootstrap/pull/254 is approved this code can updated to reflect changes in cli/vendor/github.com/drud/drud-go/secrets.
* Once updated this pull request can be merged.
* Once this repository is updated https://github.com/drud/bootstrap/pull/254 can be vendored properly using govendor and merged.
* Communicate changes necessary for moving from SECRET_EDITOR to DRUD_SECRET_EDITOR.

### CODE REVIEW

File: secrets/config.go [hunk 1]
```diff
@@ -54,18 +44,10 @@ func GetTokenDetails() (map[string]interface{}, error) {
...
-		return "", err
-	}
-	return strings.TrimSpace(string(fileBytes)), nil
-}
-
 func configEditor() {
 	// allow user to have different editor for secrets
 	// fall back to default editor
-	editor = os.Getenv("SECRET_EDITOR")
+	editor = os.Getenv("DRUD_SECRET_EDITOR")
```
Comment (beeradb) [inline secrets/config.go]: Seems like this will break backward compatibility? Maybe it's worth leaving `SECRET_EDITOR` in place as well? or maybe just holding off on this change since it seems unrelated?

Comment (rfay) [inline secrets/config.go]: I'm not actually sure why we'd need to specify a different EDITOR than $EDITOR? That's what it's for, to give things like this something to call for their editor?

Comment (cyberswat) [inline secrets/config.go]: Discussed in slack and the consensus is to move forward with DRUD_SECRET_EDITOR.  I'll update the release notes to indicate communication will be necessary.
